internal/runtime/client/client.go

```diff
@@ -250,12 +340,25 @@ func httpCall(ctx context.Context, request, response runtime.Object, opts *httpC
 	}
 	resp, err := client.Do(httpRequest)
 	if err != nil {
-		return errors.Wrap(err, "failed to perform the http call")
+		return &errCallingExtensionHandler{
+			extensionHandlerName: opts.name,
+			err:                  errors.Wrap(err, "failed to make the http call"),
+		}
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return &errCallingExtensionHandler{
+			extensionHandlerName: opts.name,
+			err:                  fmt.Errorf("non 200 response status received"),
```

Comment:
```suggestion
			err:                  errors.Errorf("non 200 response status received"),
```
I think we should use this lib consistently to get consistent errors. Let's please add the status code we got to the error message